receiver/otelarrowreceiver/config.go

```diff
@@ -0,0 +1,119 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package otelarrowreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/otelarrowreceiver"
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"path"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configgrpc"
+	"go.opentelemetry.io/collector/config/confighttp"
+	"go.opentelemetry.io/collector/confmap"
+)
+
+const (
+	// Protocol values.
+	protoGRPC = "protocols::grpc"
+	protoHTTP = "protocols::http"
+)
+
+type httpServerSettings struct {
+	*confighttp.HTTPServerSettings `mapstructure:",squash"`
+
+	// The URL path to receive traces on. If omitted "/v1/traces" will be used.
+	TracesURLPath string `mapstructure:"traces_url_path,omitempty"`
+
+	// The URL path to receive metrics on. If omitted "/v1/metrics" will be used.
+	MetricsURLPath string `mapstructure:"metrics_url_path,omitempty"`
+
+	// The URL path to receive logs on. If omitted "/v1/logs" will be used.
+	LogsURLPath string `mapstructure:"logs_url_path,omitempty"`
+}
+
+// Protocols is the configuration for the supported protocols.
+type Protocols struct {
+	GRPC  *configgrpc.GRPCServerSettings `mapstructure:"grpc"`
+	HTTP  *httpServerSettings            `mapstructure:"http"`
+	Arrow *ArrowSettings                 `mapstructure:"arrow"`
+}
+
+// ArrowSettings support configuring the Arrow receiver.
+type ArrowSettings struct {
+	// MemoryLimit is the size of a shared memory region used by
+	// all Arrow streams.  When too much load is passing through, they
+	// will see ResourceExhausted errors.
+	MemoryLimit uint64
+}
+
+// Config defines configuration for OTel Arrow receiver.
+type Config struct {
+	// Protocols is the configuration for the supported protocols, currently gRPC and HTTP (Proto and JSON).
+	Protocols `mapstructure:"protocols"`
+}
+
+var _ component.Config = (*Config)(nil)
+var _ confmap.Unmarshaler = (*Config)(nil)
+
+// Validate checks the receiver configuration is valid
+func (cfg *Config) Validate() error {
+	if cfg.GRPC == nil && cfg.HTTP == nil {
+		return errors.New("must specify at least one protocol when using the OTel Arrow receiver")
+	}
+	if cfg.Arrow != nil && cfg.GRPC == nil {
+		return errors.New("must specify at gRPC protocol when using the OTLP Arrow receiver")
```

Comment:
This message is confusing as line 64 tells us it can be either http or grpc, but this message specifically calls out grpc

Reply:
Question: https://github.com/open-telemetry/otel-arrow/issues/43

Something fairly subtle is happening here. The OTLP receiver supports two protocols and it's unnecessary in this code since the user has elected OTel Arrow, which currently requires gRPC. I've left HTTP support in for the version history, but would for many reasons like to remove it. Since I changed the default behavior relative to the core OTLP receiver, which is to enable gRPC by default, these errors become nearly impossible conditions. There is no `yaml` that can produce a nil pointer value for either of these fields, it would require a programmer explicitly setting a nil condition here. I left these statements in because of the version history, basically.

Reply:
(Is there an issue saying that `otlpreceiver` should be split into gRPC and HTTP components, I think since they use separate ports there's actually nothing shared?)

Reply:
There is no such issue that i'm aware of

Reply:
Thank you -- I take this as evidence that there is not an issue with having a multi-protocol receiver, in general. I've reworded the README in this (proposed) scaffold to explicitly call this a multi-protocol receiver. This receiver is multi-protocol in that it supports all the forms of OTLP plus the new OTel Arrow protocol. 

As an aside, [we seek this degree of integration between OTLP and Arrow because it allows senders and receivers to smoothly upgrade and downgrade between Arrow and OTLP. ](https://github.com/open-telemetry/oteps/blob/main/text/0156-columnar-encoding.md#protocol-extension-and-fallback-mechanism)